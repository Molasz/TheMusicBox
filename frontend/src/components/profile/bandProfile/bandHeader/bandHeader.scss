$color-detail: #f55110;
$max-banner-width: 600px;

.band-header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  height: 200px;

  padding-bottom: 13px;
  border-bottom: 4px solid $color-detail;
  margin-bottom: 0px;

  .band-header__logo {
    width: 200px;
  }
  .band-header__banner {
    display: none;

    flex: 1;
    justify-content: center;

    margin-left: 30px;
    .banner__img {
      max-width: $max-banner-width;
    }
  }
  .band-header__info {
    display: flex;
    flex-direction: column;
    justify-content: space-between;

    margin-left: 30px;
    text-align: end;
    .info__name {
      text-transform: uppercase;
      font-size: 60px;
      margin: 0;
    }

    .info__follow {
      width: 100%;
      display: flex;
      align-items: center;

      .contanier__icon {
        font-size: 40px;
        margin-right: 10px;
      }

      .white {
        color: white;
      }

      .orange {
        color: $color-detail;
      }

      .follow__count {
        font-size: 25px;
        margin: 0;

        .count__number {
          font-weight: bold;
        }
      }
    }
  }
}

@media (min-width: 1300px) {
  .band-header {
    .band-header__banner {
      display: block;
    }
  }
}
