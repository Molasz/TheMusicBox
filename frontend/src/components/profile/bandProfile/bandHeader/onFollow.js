import { addFollow, removeFollow } from '../../../../redux/actions/authActions';
import { toast } from 'react-toastify';

const followAlert = () =>
  toast.error('You need to login to follow bands', {
    position: 'bottom-right',
    autoClose: 5000,
    hideProgressBar: false,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    progress: undefined
  });

function onFollow(event, user, bandId, isFollowing, setIsFollowing, dispatch) {
  event.preventDefault();
  if (!(Object.keys(user).length === 0 && user.constructor === Object)) {
    if (!isFollowing) dispatch(addFollow(user._id, bandId));
    else dispatch(removeFollow(user._id, bandId));
    setIsFollowing(!isFollowing);
  } else followAlert();
  return isFollowing ? 'orange' : 'white';
}

export default onFollow;
