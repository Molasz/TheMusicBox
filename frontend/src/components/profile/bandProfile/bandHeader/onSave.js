import {
  sendBandEditInfo,
  bandEdit
} from '../../../../redux/actions/bandActions';

import store from '../../../../redux/store';

function onSave(event, bandId, data) {
  event.preventDefault();
  store.dispatch(sendBandEditInfo(bandId, data));
  store.dispatch(bandEdit({}));
}

export default onSave;
