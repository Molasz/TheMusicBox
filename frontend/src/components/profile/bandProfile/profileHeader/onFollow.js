import { addFollow, removeFollow } from '../../redux/actions/authActions';

function onFollow(event) {
  event.preventDefault();
  if (user) {
    if (!isFollowing) dispatch(addFollow(user._id, band._id));
    else dispatch(removeFollow(user._id, band._id));
    setIsFollowing(!isFollowing);
  } else alert('Login to use this feature');
  return isFollowing ? 'orange' : 'white';
}

export default onFollow;
