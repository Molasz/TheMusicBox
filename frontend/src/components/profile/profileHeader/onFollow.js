import { addFollow, removeFollow } from '../../../../redux/actions/authActions';

function onFollow(event, userId, bandId, dispatch) {
  event.preventDefault();
  /* if (user) {
    if (!isFollowing) dispatch(addFollow(userId, bandId));
    else dispatch(removeFollow(userId, bandId));
    setIsFollowing(!isFollowing);
  } else alert('Login to use this feature');
  return isFollowing ? 'orange' : 'white'; */
}

export default onFollow;
