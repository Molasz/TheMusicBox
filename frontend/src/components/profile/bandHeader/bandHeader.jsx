import React, { useState } from 'react';
import { connect } from 'react-redux';

import './bandHeader.scss';

import onFollow from './onFollow';

import Star from '@material-ui/icons/Grade';

function BandHeader({ logo, banner, name, followers, match, dispatch }) {
  const [followIconClass, setFollowIconClass] = useState('');

  const infoFollow = (
    <div className='follow__container'>
      <Star
        className={`contanier__icon ${followIconClass}`}
        id='follow'
        onClick={(event) => onFollow(event, 0, match.params.bandId, dispatch)}
      />
      <p className='follow__count'> {followers} Followers</p>
    </div>
  );

  return (
    <>
      <img src={logo} alt='Logo' className='top__logo' />
      <img src={banner} alt='Banner' className='top__banner' />
      <div className='top__info'>
        <strong className='info__name'>{name}</strong>
        <div className='info__follow'>{infoFollow}</div>
      </div>
    </>
  );
}

function mapStateToProps(state) {
  return { followers: state.bandReducer.bandFollowers };
}

export default connect(mapStateToProps)(BandHeader);
export { BandHeader };
