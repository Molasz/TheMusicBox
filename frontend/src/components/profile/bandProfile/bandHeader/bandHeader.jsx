import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';

import './bandHeader.scss';

import onFollow from './onFollow';

import Star from '@material-ui/icons/Grade';
import Gear from '@material-ui/icons/Settings';
import Save from '@material-ui/icons/Save';

function BandHeader({ logo, banner, name, bandId, user, dispatch, followers }) {
  const [isFollowing, setIsFollowing] = useState(null);

  useEffect(() => {
    if (
      isFollowing === null &&
      !(Object.keys(user).length === 0 && user.constructor === Object)
    ) {
      setIsFollowing(user.following.some((element) => element._id === bandId));
    }
  }, [bandId, isFollowing, user]);

  const followIconClass = isFollowing ? 'orange' : 'white';

  return (
    <section className='band-header'>
      <img src={logo} alt='Logo' className='band-header__logo' />
      <div className='band-header__banner'>
        <img src={banner} alt='Banner' className='banner__img' />
      </div>
      <div className='band-header__info'>
        <strong className='info__name'>{name}</strong>
        <div className='info__bottom'>
          <div className='bottom__follow'>
            <Star
              className={`contanier__icon ${followIconClass}`}
              id='follow'
              onClick={(event) =>
                onFollow(
                  event,
                  user,
                  bandId,
                  isFollowing,
                  setIsFollowing,
                  dispatch
                )
              }
            />
            <p className='follow__count'>
              <span className='count__number'>{followers}</span> Followers
            </p>
          </div>
          <div className='bottom__edit'>
            <Gear className='edit__gear' onClick={(event) => {}} />
            {true && <Save className='edit__save' onClick={(event) => {}} />}
          </div>
        </div>
      </div>
    </section>
  );
}

function mapStateToProps(state) {
  return {
    followers: state.bandReducer.bandFollowers,
    user: state.authReducer.user
  };
}

export default connect(mapStateToProps)(BandHeader);
export { BandHeader };
