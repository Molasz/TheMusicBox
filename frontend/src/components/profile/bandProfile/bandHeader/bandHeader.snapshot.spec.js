import React from 'react';
import renderer from 'react-test-renderer';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });

import onFollow from './onFollow';
jest.mock('./onFollow');

import { BandHeader } from './bandHeader';

describe('BandHeader snapshot', () => {
  it('Should match ', () => {
    const logo = 'logo';
    const banner = 'banner';
    const name = 'name';
    const bandId = 'bandId';
    const user = { following: ['1'] };
    const followers = 0;

    const tree = renderer.create(
      <BandHeader
        logo={logo}
        banner={banner}
        name={name}
        bandId={bandId}
        user={user}
        followers={followers}
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Should call onFollow when click', () => {
    const document = shallow(<BandHeader />);
    const button = document.find('#follow');
    button.simulate('click', { preventDefault: () => {} });

    expect(onFollow.call).truthy;
  });
});
