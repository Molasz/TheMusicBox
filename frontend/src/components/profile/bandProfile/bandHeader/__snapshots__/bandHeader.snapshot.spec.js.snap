// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BandHeader snapshot Should match  1`] = `
<section
  className="band-header"
>
  <img
    alt="Logo"
    className="band-header__logo"
    src="logo"
  />
  <div
    className="band-header__banner"
  >
    <img
      alt="Banner"
      className="banner__img"
      src="banner"
    />
  </div>
  <div
    className="band-header__info"
  >
    <strong
      className="info__name"
    >
      name
    </strong>
    <div
      className="info__follow"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root contanier__icon white"
        focusable="false"
        id="follow"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
      <p
        className="follow__count"
      >
        <span
          className="count__number"
        >
          0
        </span>
         Followers
      </p>
    </div>
  </div>
</section>
`;
